cni/pkg/cmd/root.go

```diff
@@ -161,7 +174,11 @@ func init() {
 	registerBooleanParameter(constants.SkipTLSVerify, false, "Whether to use insecure TLS in kubeconfig file")
 	registerIntegerParameter(constants.MonitoringPort, 15014, "HTTP port to serve prometheus metrics")
 	registerStringParameter(constants.LogUDSAddress, "/var/run/istio-cni/log.sock", "The UDS server address which CNI plugin will copy log output to")
+	registerStringParameter(constants.CNIEventAddress, "/var/run/istio-cni/pluginevent.sock",
```

Comment:
Required for the  netns, actually, and to block pod readiness until CNI is done, which is a requirement.

We can annotate, which would trigger a new event received by the informer - a pod update, as well as the pod add event it already gets in this case. However, we have no ordering guarantees there, nor do we have any guarantees that we will get those events thru k8s before the pod is actually scheduled - the K8S API is not a reliable or timely source of information for new pod spawns, and relying on it can create race conditions with the container runtime.

The CNI is the first, best, and most reliable source for what we need here.

Certainly open to later iteration that meets the requirements tho.

Reply:
Yep, that could work, though it would be a bit baroque to reason about in its own way. There are other design approaches certainly.

@yuval-k in other comments has mentioned that one way we could deprivilege `istio-cni` and drop some of the node kernel FS mounts is by passing the container FD thru this UDS channel, which would not be possible if we removed the UDS channel.

That being said, if we want to change the design here, I'd rather do that in a discrete followup and have a proper discussion on the tradeoffs.